deepchem/feat/base_classes.py

```diff
@@ -73,27 +117,56 @@ def _featurize_complex(self, mol_pdb, complex_pdb):
     raise NotImplementedError('Featurizer is not defined.')
 
 
-class Featurizer(object):
-  """
-  Abstract class for calculating a set of features for a molecule.
+class MolecularFeaturizer(object):
+  """Abstract class for calculating a set of features for a
+  molecule.
 
-  Child classes implement the _featurize method for calculating features
-  for a single molecule.
+  The defining feature of a `MolecularFeaturizer` is that it
+  uses SMILES strings and RDKIT molecule objecgs to represent
+  small molecules. All other featurizers which are subclasses of
+  this class should plan to process input which comes as smiles
+  strings or RDKIT molecules. 
+
+  Child classes need to implement the _featurize method for
+  calculating features for a single molecule.
+
+  Note
+  ----
+  In general, subclasses of this class will require RDKit to be installed.
   """
 
   def featurize(self, mols, verbose=True, log_every_n=1000):
-    """
-    Calculate features for molecules.
+    """Calculate features for molecules.
 
     Parameters
     ----------
     mols : iterable
-        RDKit Mol objects.
+        RDKit Mol, or SMILES string, or filename for
+        mol2/sdf/pdb/pdbqt file.
```

Comment:
I think this `featurize ` should receive only rdkit mol object.
Converting SMILES to mol object is responsible for data_loader function.

Reply:
I don't see where it actually handles filenames.  That would be ambiguous in any case: if it gets a string, how does it decide whether to interpret it as a SMILES string or a filename?

Reply:
That's some cruft there about the filenames. Will filter out!

I think handling SMILES is very useful though. The main reason is it facilitates experimentation. Let's say I have a pandas dataframe with a column `col` of smiles. I can then do really interesting things like

```
X = featurizer(col)
```

if `featurizer` is the featurizer we're using. Right now, we have to do an intermediate processing step like
```
mols = [Chem.MolFromSmiles(s) for s in col]
X = featurizer(mols)
```
The docs also become a lot cleaner. We can give examples like
```
f = CircularFingerprint()
x = f("CC")
```

Reply:
I totally agree!  Having MolecularFeaturizer that can automatically convert SMILES strings to mols will be very convenient.

Reply:
I could understand the usecase! If this featurizer takes only RDKit Mol object or SMILES, I agree!